source/plugins/go/src/log_loss_monitoring.go

```diff
@@ -0,0 +1,303 @@
+structures than are in the public count, it will pick five containers from its internal data structures and write their
+IDs to a shared chanel (deleted_containers_query) (in a non-blocking way, if the channel is full then retry later)
+
+3. The FW thread will read the containerIds form deleted_containers_query, then if they are not on disk then write the
+containerIds back to a second shared channel (deleted_containers_response)
+
+4. The Main thread will read from deleted_containers_response (also non-blocking) and delete any recieved
+containerIDs from it's own tables. (the delay is so that any logs from a freshly deleted container will have time to go through fbit)
+
+Motivation: with this method, the Main thread should store about 1.2x as many containers as are on disk (assuming it queries about deleted containers
+in a evenly distributed way). If more than 1/5 of the containers in the Main thread's data structures are deleted, then on average query more
+than one deleted containerId. That will cause the number of containers the Main thread is storing to shrink.
+*/
+
+// Main thread managed data structures:
+// var m_container_to_arr_index = make(map[string]int64)
+var m_bytes_logged_storage AddressableMap
+
+var m_deletion_query_index = 0
+
+// File Watcher managed data structures
+// (FW = File Watcher)
+type FwRecord struct {
+	unrotated_bytes    int64
+	rotated_bytes      int64
+	existing_log_files []string
+}
+
+var FW_records map[string]FwRecord // identifier -> file names
+// var FW_bytes_logged_rotated = make(map[string]int64)   // identifier -> count
+// var FW_bytes_logged_unrotated = make(map[string]int64) // identifier -> count
+// var current_iteration_count int64 = 0
+var num_containers_on_disk int32
+
+// Shared objects
+var deleted_containers_query chan string    // There might be a lot more than 5 containers  of 10 chosen arbitrairly
+var deleted_containers_response chan string // capacity of 10 chosen arbitrairly
+
+var process_logs_mut sync.Mutex
+var read_disk_mut sync.Mutex
+var enabled bool
+
+var disabled_namespaces map[string]bool
+
+var log_loss_logger *log.Logger
+
+func init() {
+	init_log_loss_monitoring_globals()
+}
+
+// This function exists for unit tests, it lets each test reset all global variables at the start of each test
+func init_log_loss_monitoring_globals() {
+	m_bytes_logged_storage = Make_AddressableMap()
+
+	m_deletion_query_index = 0
+
+	FW_records = make(map[string]FwRecord) // identifier -> file names
+	// var FW_bytes_logged_rotated = make(map[string]int64)   // identifier -> count
+	// var FW_bytes_logged_unrotated = make(map[string]int64) // identifier -> count
+	// var current_iteration_count int64 = 0
+	num_containers_on_disk = 0
+
+	// Shared objects
+	deleted_containers_query = make(chan string, 5)    // There might be a lot more than 5 containers  of 10 chosen arbitrairly
```

Comment:
probable having constant for the capacity would make easy to change if needed.

Reply:
done